src/JustSaying/AwsTools/MessageHandling/SnsTopicByName.cs

```diff
@@ -85,49 +85,49 @@ public override async Task<bool> ExistsAsync()
             return false;
         }
 
-        public async Task<bool> CreateAsync()
+        public async Task CreateAsync()
         {
             try
             {
                 var response = await Client.CreateTopicAsync(new CreateTopicRequest(TopicName))
                     .ConfigureAwait(false);
 
-                if (!string.IsNullOrEmpty(response.TopicArn))
+                if (string.IsNullOrEmpty(response.TopicArn))
                 {
-                    Arn = response.TopicArn;
-                    _log.LogInformation("Created topic '{TopicName}' on ARN '{Arn}'.", TopicName, Arn);
-                    return true;
+                    _log.LogError("Failed to create topic or obtain ARN for topic {TopicName}", TopicName);
+                    throw new Exception($"Failed to create topic or receive ARN '{TopicName}'.");
```

Comment:
Why do you want to have a different error message?
```suggestion
                    throw new Exception($"Failed to create or obtain ARN for topic '{TopicName}'.");
```

Reply:
Fixed it up so they're the same 👍